src/ring/middleware/oauth2.clj

```diff
@@ -37,10 +37,13 @@
   (= (get-in request [:session ::state])
      (get-in request [:query-params "state"])))
 
-(defn- format-access-token [{{:keys [access_token expires_in]} :body :as r}]
+(defn- format-access-token
+  [{{:keys [access_token expires_in refresh_token id_token]} :body :as r}]
   (-> {:token access_token}
       (cond-> expires_in (assoc :expires (-> expires_in time/seconds
-                                                        time/from-now)))))
+                                                      time/from-now))
+              refresh_token (assoc :refresh_token refresh_token)
```

Comment:
The key in the session shouldn't include underscores, so `:refresh-token` rather than `:refresh_token`.